### ISSUE TITLE
config_flow.py line 815: `hass_url` undefined

### ISSUE THREAD

**Issue by danielbrunt57** [alandtse/alexa_media_player#2408]
Sorry everyone but I just had an automatic re-auth situation which triggered the `update_schema_defaults `and unfortunately there was a pre-existing `self.config.pop("hass_url")` at line 621 in `async def _test_login` making `hass_url `unavailable after that.

I removed lines 621-622, restarted and the problem was resolved.

Issuing a new PR for it...

**Comment by LorenzoRogai** [comment]
just tested this by changing code manually but still error

<img width="920" alt="image" src="https://github.com/user-attachments/assets/17e84c04-ec0c-45e4-9969-6e8f7403f0e3">

**Comment by bkr1969** [comment]
Any ETA to fix this?  I can "install" the integration, but none of my devices are listed now. Reconfiguring just causes error.

**Comment by gabriele-latino** [comment]
Hi, same issue here on 2024.8

**Comment by mazdarx2001** [comment]
Same here, does anyone know if this is an issue on all alexa Media integrations under the new 2024.8? or just a small number?

**Comment by Impedanz** [comment]
Same Error here...

**Comment by tksly92** [comment]
Same problem here

**Comment by bkr1969** [comment]
Not sure how you deleted line 622.  There's an open ```{``` on that line.  Am I missing something?

**Comment by LorenzoRogai** [comment]
hass_url error is fixable by changing the line to

```
CONF_HASS_URL, default=self.config.get(CONF_HASS_URL, DEFAULT_HASS_URL)
```

if you are in US then this will be enough. For the others countries alexapy will also need a fix https://github.com/alandtse/alexa_media_player/issues/2407

**Comment by bkr1969** [comment]
Sorry.  That doesn't fix the entire issue.  The integration loads, but no devices are listed. A reconfigure still errors.

**Comment by danielbrunt57** [comment]
> Sorry. That doesn't fix the entire issue. The integration loads, but no devices are listed. A reconfigure still errors.

as I am seeing also,,,
```
aiohttp.client_exceptions.InvalidUrlClientError: /errors/validateCaptcha
```

**Comment by fbnksr** [comment]
> hass_url error is fixable by changing the line to
> 
> ```
> CONF_HASS_URL, default=self.config.get(CONF_HASS_URL, DEFAULT_HASS_URL)
> ```
> 
> if you are in US then this will be enough. For the others countries alexapy will also need a fix #2407

This fixed the issue for me! Thanks!

**Comment by fbnksr** [comment]
> hass_url error is fixable by changing the line to
> 
> 
> 
> ```
> 
> CONF_HASS_URL, default=self.config.get(CONF_HASS_URL, DEFAULT_HASS_URL)
> 
> ```
> 
> 
> 
> if you are in US then this will be enough. For the others countries alexapy will also need a fix https://github.com/alandtse/alexa_media_player/issues/2407

While this resolved the ‘failed to setup’ issue and everything is functioning again, some of my devices, like my Amazon smart plug, are no longer appearing. Any ideas on how to fix this?
